credmark/dto/encoder.py

```diff
@@ -17,9 +17,9 @@ def default(self, o):
         return json.JSONEncoder.default(self, o)
 
 
-def json_dump(obj, fp):  # pylint: disable=invalid-name
+def json_dump(obj, fp, post_proc=lambda x: x, **b):  # pylint: disable=invalid-name
     """Dump an obj that may contain embedded DTOs to json"""
-    return json.dump(obj, fp, cls=PydanticJSONEncoder)
+    return post_proc(json.dump(obj, fp, cls=PydanticJSONEncoder, **b))
```

Comment:
Sure. I have changed the parameter name to `json_dump_args`. I also moved the string processing function to `json_dumps`.

```
def json_dump(obj, handle, **json_dump_args):
    """Dump an obj that may contain embedded DTOs to json"""
    return json.dump(obj, handle, cls=PydanticJSONEncoder, **json_dump_args)

def json_dumps(obj, format_json=lambda x: x, **json_dump_args):
    """Dump an obj that may contain embedded DTOs to json string"""
    return format_json(json.dumps(obj, cls=PydanticJSONEncoder, **json_dump_args))
```

Reply:
Because json.dump multi-line string as quoted. The trace log looks like below. Replacement from quoted `\\n` to `\n` is needed.

Original
```
"trace": "  File \"/home/yangye/dev/credmark/credmark-models-py/../credmark-model-framework-py/credmark/types/data/contract.py\", line 245, in proxy_for\n    raise ModelDataError(f'Unable to retrieve abi for proxy for {self.address}')\n  File \"/home/yangye/dev/credmark/credmark-models-py/../credmark-model-framework-py/credmark/types/data/contract.py\", line 141, in _load\n    if self.proxy_for is not None:\n  File \"/home/yangye/dev/credmark/credmark-models-py/../credmark-model-framework-py/credmark/types/data/token.py\", line 98, in _load\n    super()._load()\n  File \"/home/yangye/dev/credmark/credmark-models-py/../credmark-model-framework-py/credmark/types/data/contract.py\", line 167, in proxy_for\n    self._load()\n  File \"/home/yangye/dev/credmark/credmark-models-py/./models/credmark/tokens/token.py\", line 91, in run\n    _ = input.proxy_for\n"
```

With the string process lambda:
```
 File "/home/yangye/dev/credmark/credmark-models-py/../credmark-model-framework-py/credmark/types/data/contract.py", line 245, in proxy_for
    raise ModelDataError(f'Unable to retrieve abi for proxy for {self.address}')
  File "/home/yangye/dev/credmark/credmark-models-py/../credmark-model-framework-py/credmark/types/data/contract.py", line 141, in _load
    if self.proxy_for is not None:
  File "/home/yangye/dev/credmark/credmark-models-py/../credmark-model-framework-py/credmark/types/data/token.py", line 98, in _load
    super()._load()
  File "/home/yangye/dev/credmark/credmark-models-py/../credmark-model-framework-py/credmark/types/data/contract.py", line 167, in proxy_for
    self._load()
  File "/home/yangye/dev/credmark/credmark-models-py/./models/credmark/tokens/token.py", line 91, in run
    _ = input.proxy_for
```

Reply:
But whatever is calling `json_dumps()` can run whatever processing they need on the output. 

`format_json_whatever(json_dumps(obj))`

Reply:
ok. I have remove the lambda.